src/lib/result.test.js

```diff
@@ -0,0 +1,52 @@
+// @flow
+import { ok, error, map, withDefault } from "./result";
+
+describe("ok", () => {
+  it("creates an ok result", () => {
+    const result = ok("test");
+    expect(result.ok).toEqual(true);
+    if (result.ok) {
```

Comment:
This this `if` necessary because of flow?

Reply:
yeah, a bit annoying because I had to wrap all the Result value/error assertions with this.